src/cpp/src/sampler.cpp

```diff
@@ -752,13 +888,14 @@ SamplerOutput Sampler::sample(const std::vector<SequenceGroup::Ptr> & sequence_g
     size_t vocab_size = logits_shape[2];
 
     SamplerOutput sampler_output;
+    std::unordered_map<SequenceGroup::Ptr, std::future<SequenceGroupSamplingInfo>> sg_sampling_future_map;
```

Comment:
Looks like potentially dangerous place... I mean to use `shared_ptr` as a key in `unordered_map` according hash using... To replace it by `seq_group id`?